applications/ShapeOptimizationApplication/python_scripts/algorithm_penalized_projection.py

```diff
@@ -94,96 +114,85 @@ def RunOptimizationLoop( self ):
-        self.DampingUtilities.DampNodalVariable( OBJECTIVE_SENSITIVITY )
-        self.DampingUtilities.DampNodalVariable( CONSTRAINT_SENSITIVITY )
-
-    # --------------------------------------------------------------------------
-    def __computeShapeUpdate( self ):
+    def __computeShapeUpdate(self):
         self.__mapSensitivitiesToDesignSpace()
-        constraint_value = self.Communicator.getStandardizedValue( self.onlyConstraintId )
-        if self.__isConstraintActive( constraint_value ):
+        constraint_value = self.Communicator.getStandardizedValue(self.only_con["identifier"].GetString())
+        if self.__isConstraintActive(constraint_value):
             self.OptimizationUtilities.ComputeProjectedSearchDirection()
-            self.OptimizationUtilities.CorrectProjectedSearchDirection( constraint_value )
+            self.OptimizationUtilities.CorrectProjectedSearchDirection(constraint_value)
         else:
             self.OptimizationUtilities.ComputeSearchDirectionSteepestDescent()
         self.OptimizationUtilities.ComputeControlPointUpdate()
         self.__mapDesignUpdateToGeometrySpace()
 
+        if self.isDampingSpecified:
+            self.DampingUtilities.DampNodalVariable(SHAPE_UPDATE)
+
     # --------------------------------------------------------------------------
-    def __mapSensitivitiesToDesignSpace( self ):
-        self.Mapper.MapToDesignSpace( OBJECTIVE_SENSITIVITY, MAPPED_OBJECTIVE_SENSITIVITY )
-        self.Mapper.MapToDesignSpace( CONSTRAINT_SENSITIVITY, MAPPED_CONSTRAINT_SENSITIVITY )
+    def __mapSensitivitiesToDesignSpace(self):
+        self.Mapper.MapToDesignSpace(DF1DX, DF1DX_MAPPED)
+        self.Mapper.MapToDesignSpace(DC1DX, DC1DX_MAPPED)
 
     # --------------------------------------------------------------------------
-    def __isConstraintActive( self, constraintValue ):
-        if self.typeOfOnlyConstraint == "=":
+    def __isConstraintActive(self, constraintValue):
+        if self.only_con["type"].GetString() == "=":
             return True
         elif constraintValue > 0:
             return True
         else:
             return False
 
     # --------------------------------------------------------------------------
-    def __mapDesignUpdateToGeometrySpace( self ):
-        self.Mapper.MapToGeometrySpace( CONTROL_POINT_UPDATE, SHAPE_UPDATE )
+    def __mapDesignUpdateToGeometrySpace(self):
+        self.Mapper.MapToGeometrySpace(CONTROL_POINT_UPDATE, SHAPE_UPDATE)
 
     # --------------------------------------------------------------------------
-    def __dampShapeUpdate( self ):
-        self.DampingUtilities.DampNodalVariable( SHAPE_UPDATE )
+    def __logCurrentOptimizationStep(self):
+        additional_values_to_log = {}
+        additional_values_to_log["step_size"] = self.algorithm_settings["line_search"]["step_size"].GetDouble()
+        additional_values_to_log["correction_scaling"] = self.algorithm_settings["correction_scaling"].GetDouble()
```

Comment:
does this log the uupdated scaling if the `adaptive_correction_scaling` is used?

Reply:
yes